File: 2559-count-vowel-strings-in-ranges/2559-count-vowel-strings-in-ranges.java
class Solution {
    public int[] vowelStrings(String[] words, int[][] queries) {
        int[] bool = new int[words.length+1];
        int[] ans = new int[queries.length];
        int count = 0;
        bool[0] = 0;
        int boolIndex = 1;
        for(int i=0;i<words.length;i++){
            String st = words[i];
            int firstChar = st.charAt(0);
            int lastChar = st.charAt(st.length()-1);
            if((firstChar==('a') || firstChar==('e') || firstChar==('i') || firstChar==('o') || firstChar==('u')) && (lastChar==('a') || lastChar==('e') || lastChar==('i') || lastChar==('o') || lastChar==('u'))){
                count++;
                bool[boolIndex] = count;
                boolIndex++;
            }else{
                bool[boolIndex] = count;
                boolIndex++;
            }
            
        }

        for(int i=0;i<queries.length;i++){
            
            int first = queries[i][0];
            int last = queries[i][1];
            int diff = bool[last+1]-bool[first];
            ans[i] = diff;
        }
        
        return ans;
    }
}

File: 2559-count-vowel-strings-in-ranges/2559-count-vowel-strings-in-ranges.py
class Solution:
    def vowelStrings(self, words: List[str], queries: List[List[int]]) -> List[int]:
        prefix = [0]
        ans = []
        count = 0
        for i in range(len(words)):
            firstChar = words[i][0]
            lastChar = words[i][-1]
            if (firstChar in "aeiou" and lastChar in "aeiou"):
                count=count+1
                prefix.append(count)
            else:
                prefix.append(count)
        print(prefix)
        for i in range(len(queries)):
            first = queries[i][0]
            last = queries[i][1]
            diff = prefix[last+1]-prefix[first]
            ans.append(diff)
        return ans
